spec/controllers/omniauth_callbacks_controller_spec.rb

```diff
@@ -0,0 +1,74 @@
+require 'spec_helper'
+
+describe OmniauthCallbacksController do
+
+  fixtures :users
+
+  describe '#login_dot_gov' do
+    before do
+      mock_user_auth('12345', 'test@gsa.gov')
+      request.env['omniauth.auth'] = OmniAuth.config.mock_auth[:login_dot_gov]
+      get :login_dot_gov
+    end
+
+    let(:user) { User.find_by(email: 'test@gsa.gov') }
+
+    it { is_expected.to redirect_to(admin_home_page_path) }
+
+    context 'when the new user info is saved' do
+      before do
+        get :login_dot_gov
+      end
+
+      it 'saves the email' do
+        expect(user.email).to(eq 'test@gsa.gov')
+      end
+      it 'saves the uid' do
+        expect(user.uid).to(eq '12345')
+      end
+    end
+
+    context 'when the existing user with no uid info is saved' do
+      before do
+        mock_user_auth('12345', 'user_with_out_uid@fixtures.org')
+        get :login_dot_gov
+      end
+
+      let(:user) { User.find_by(email: 'user_with_out_uid@fixtures.org') }
+
+      it 'saves the email' do
+        expect(user.email).to(eq 'user_with_out_uid@fixtures.org')
+      end
+      it 'saves the uid' do
+        expect(user.uid).not_to be_nil
```

Comment:
This spec is passing for the wrong reasons. Because the UID in the fixture is actually the string `'nil'` (see notes in the fixture), this spec is passing because the `user.uid` is still the string `'nil'`. The login is not affecting that user record because it's using the auth object for `test@gsa.gov` (see above). The correct test (which will fail until the correct auth object is used) would be:
```ruby
expect(user.uid).to eq '11111'
```